linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_clients.js

```diff
@@ -42,17 +53,66 @@ var RouterClient = (function() {
     return summary;
   }
 
-  return function (metricsCollector, routers, client, $clientEl, routerName, clientTemplate) {
+  function timeseriesParams() {
+    return {
+      strokeStyle: colorLookup[colorIdx++ % colorLookup.length], //TODO: #242 per client color scheme
+      lineWidth: 2
+    };
+  };
+
+  function initializeChart($chartEl, latencyKeys) {
+    var $canvas = $("<canvas id='canvas' height='181'></canvas>");
+    $chartEl.append($canvas);
+
+    var chart = new UpdateableChart(
+      {
+        minValue: 0,
+        grid: {
+          strokeStyle: '#878787',
+          verticalSections: 1,
+          millisPerLine: 10000,
+          borderVisible: false
+        },
+        labels: {
+          fillStyle: '#878787',
+          fontSize: 12,
+          precision: 0
+        },
+        millisPerPixel: 60
+      },
+      $canvas[0],
+      function() {
+        return $chartEl.width();
+      },
+      timeseriesParams
+    );
+    var desiredLatencyMetrics = _.map(latencyKeys, function(metric) {
+      return {
+        name: metric,
+        color: ""
+      }
+    });
+    chart.setMetrics(desiredLatencyMetrics, timeseriesParams, true);
+
+    return chart;
+  }
+
+  return function (metricsCollector, routers, client, $metricsEl, routerName, clientTemplate, $chartEl, colors) {
     template = clientTemplate;
-    var metricDefinitions = getMetricDefinition(routerName, client.label);
-    var latencyKeys = Query.filter(/^request_latency_ms\..*/, _.keys(client.metrics));
+    colorLookup = colors;
+    var metricDefinitions = getMetricDefinitions(routerName, client.label);
+    var latencyKeys = Query.filter(/^request_latency_ms\.(max|min|p9990|p99|p95|p50).*/, _.keys(client.metrics));
```

Comment:
I don't think you need the wildcard match after the histogram value here, since those should be the end of the string that you're trying to match against.  As written, this regex will also match on p9999.  It would be better to include a trailing anchor to avoid that.

```
> Query.filter(/^request_latency_ms\.(max|min|p9990|p99|p95|p50).*/, [{name: "request_latency_ms.p9999"}])[0];
<- Object {name: "request_latency_ms.p9999"}
> Query.filter(/^request_latency_ms\.(max|min|p9990|p99|p95|p50)$/, [{name: "request_latency_ms.p9999"}])[0];
<- undefined
```